packages/web-components/fast-foundation/src/accordion-item/accordion-item.template.ts

```diff
@@ -1,56 +1,55 @@
-import { html, ref } from "@microsoft/fast-element";
-import type { ViewTemplate } from "@microsoft/fast-element";
-import { endSlotTemplate, startSlotTemplate } from "../patterns/start-end.js";
-import type { FoundationElementTemplate } from "../foundation-element/foundation-element.js";
-import type { AccordionItem, AccordionItemOptions } from "./accordion-item.js";
+import { ElementViewTemplate, html, ref } from "@microsoft/fast-element";
+import { createEndSlotTemplate, createStartSlotTemplate } from "../patterns/start-end.js";
+import type { AccordionItemOptions, FoundationAccordionItem } from "./accordion-item.js";
 
 /**
- * The template for the {@link @microsoft/fast-foundation#(AccordionItem:class)} component.
+ * The template for the {@link @microsoft/fast-foundation#(FoundationAccordionItem:class)} component.
  * @public
  */
-export const accordionItemTemplate: FoundationElementTemplate<
-    ViewTemplate<AccordionItem>,
-    AccordionItemOptions
-> = (context, definition) => html`
-    <template class="${x => (x.expanded ? "expanded" : "")}">
-        <div
-            class="heading"
-            part="heading"
-            role="heading"
-            aria-level="${x => x.headinglevel}"
-        >
-            <button
-                class="button"
-                part="button"
-                ${ref("expandbutton")}
-                aria-expanded="${x => x.expanded}"
-                aria-controls="${x => x.id}-panel"
-                id="${x => x.id}"
-                @click="${(x, c) => x.clickHandler(c.event as MouseEvent)}"
+export function createAccordionItemTemplate(
```

Comment:
Is the thought with this that the template itself will evolve in the coming week to migrate something resembling the builder? IE, we have functions to instantiate `start()` and `end()`, or are we thinking that this will be a singular template such as this where those are passed via "options"? 

If this will represent the "builder" nature, then I think `create` makes sense; however, if it's simply a single function which returns HTML like this, then I'd prefer to avoid that terminology because we're just instantiating a fixed thing...

Reply:
I'm thinking of this in two phases. This first pass is just a basic template that creates a fixed template. Then in a future phase, this will be a function that uses the builder to set up the default template and return that. The output of the function will be the same in both cases. The internal implementation will change to using a builder.

For example:

```ts
export function createAccordionItem(options: AccordionItemOptions) {
    return AccordionItemAnatomy.define()
        .anatomy(item =>
            item.heading(heading =>
                heading
                    .button()
                    .startSlot({ fallback: options.start || "" })
                    .endSlot({ fallback: options.end || "" })
                    .icons(icons =>
                        icons
                            .expanded({ fallback: options.expandedIcon || "" })
                            .collapsed({ fallback: options.collapsedIcon || "" })
                    )
            )
        )
        .build();
}
```

The above outputs the same as the HTML template we have today, but it uses the builder internally instead. Those who want to customize further would use the builder directly and not use this function at all.

Reply:
FYI I'm using the `create` naming scheme because it actually does create a new template every time you call the function. I'm not strongly tied to this though. I just don't want to change the names after this pass, regardless of how the internal implementation changes.